### PR TITLE
Bump to latest Kafka clients 3.7

### PR DESCRIPTION
**Pull request by pierDipi**
As per title

### CODE REVIEW

File: data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/RecordDispatcherImpl.java [hunk 1]
```diff
@@ -362,7 +362,7 @@ private ConsumerRecordContext maybeDeserializeValueFromHeaders(ConsumerRecordCon
         // headers even though the record is a valid CloudEvent.
         logDebug("Value is null", recordContext.getRecord());
         final var value = cloudEventDeserializer.deserialize(
-                recordContext.getRecord().topic(), recordContext.getRecord().headers(), null);
+                recordContext.getRecord().topic(), recordContext.getRecord().headers(), (byte[]) null);
```
Comment (matzew) [inline data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/RecordDispatcherImpl.java:365]: this is b/c of here, right? https://github.com/apache/kafka/pull/12545/

**Review by Cali0707 — APPROVED** [review]
/lgtm

**Review by Cali0707 — APPROVED** [review]
/lgtm

**Pull request merged**